jupyter_client/manager.py

```diff
@@ -403,11 +403,19 @@ def interrupt_kernel(self):
         platforms.
         """
         if self.has_kernel:
-            if sys.platform == 'win32':
-                from .win_interrupt import send_interrupt
-                send_interrupt(self.kernel.win32_interrupt_event)
-            else:
-                self.signal_kernel(signal.SIGINT)
+            interrupt_mode = self.kernel_spec.interrupt_mode
+            if interrupt_mode == 'signal':
+                if sys.platform == 'win32':
+                    from .win_interrupt import send_interrupt
+                    send_interrupt(self.kernel.win32_interrupt_event)
+                    self._send_signal_message(signal.SIGINT)
```

Comment:
I don't think the `_send_signal_message` applies anymore. Might be left over from the initial commit?

Reply:
You're right, fixed :)